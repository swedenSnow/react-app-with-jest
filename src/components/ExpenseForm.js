import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

// const date = new Date();

const now = moment();
// console.log(now.format("MMM Do, YYYY"));

class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense
                ? (props.expense.amount / 100).toString()
                : '',
            createdAt: props.expense
                ? moment(props.expense.createdAt)
                : moment(),
            calenderFocused: false,
            error: '',
        };
    }

    onDescriptionChange = e => {
        const description = e.target.value;
        this.setState(() => ({
            description,
        }));
    };

    onNoteChange = e => {
        const note = e.target.value;
        // Could also use e.persist() instead of a const varible
        // e.persist();
        // this.setState(() => ({ note: e.target.value }));
        this.setState(() => ({
            note,
        }));
    };

    onAmountChange = e => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    };

    onDateChange = createdAt => {
        if (createdAt) {
            this.setState(() => ({
                createdAt,
            }));
        }
    };

    onFocusChange = ({ focused }) => {
        this.setState(() => ({
            calenderFocused: focused,
        }));
    };

    onSubmit = e => {
        e.preventDefault();

        if (!this.state.description) {
            this.setState(() => ({
                error: 'Please enter a valid description ',
            }));
        } else if (!this.state.amount) {
            this.setState(() => ({
                error: "Please enter a valid amount. Nothing's free!",
            }));
        } else {
            this.setState(() => ({
                error: '',
            }));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note,
            });
        }
        // if (!this.state.description || !this.state.amount) {
        // this.setState(() => ({
        //   error: "Please enter a valid desc and amount",
        // }));
        // } else {
        //   this.setState(() => ({
        //     error: "Submitted",
        //   }));
        // }
    };

    render() {
        const {
            description,
            error,
            note,
            amount,
            createdAt,
            calenderFocused,
        } = this.state;
        return (
            <form className="form" onSubmit={this.onSubmit}>
                {error && (
                    <p className="form__error">
                        <strong>{error}</strong>
                    </p>
                )}
                <input
                    type="text"
                    className="text-input"
                    placeholder="Description"
                    autoFocus
                    value={description}
                    onChange={this.onDescriptionChange}
                />
                <input
                    type="text"
                    className="text-input"
                    placeholder="Amount"
                    value={amount}
                    onChange={this.onAmountChange}
                />
                <SingleDatePicker
                    date={createdAt}
                    onDateChange={this.onDateChange}
                    focused={calenderFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
                <textarea
                    className="textarea"
                    placeholder="Add a note for your expense (optional)"
                    value={note}
                    onChange={this.onNoteChange}
                />
                <div>
                    <button className="button--link">Save Expense</button>
                </div>
            </form>
        );
    }
}

export default ExpenseForm;
