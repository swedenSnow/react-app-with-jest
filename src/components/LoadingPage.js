import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <img className="loader__image" src="images/loading3.gif" />
  </div>
);

export default LoadingPage;
