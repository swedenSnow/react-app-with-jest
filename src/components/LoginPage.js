import React from 'react';
import { connect } from 'react-redux';

import { startLogin } from '../actions/auth';

//stateless funtional component
export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Keep track of your Budget</h1>
      <p>It's time to get your expenses under control</p>

      <button className="button" onClick={startLogin}>
        Sign in with
        <span className="btn__icon" />
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
