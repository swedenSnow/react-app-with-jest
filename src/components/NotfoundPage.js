import React from 'react';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        404 - <Link to="/">Go home</Link>
    </div>
);

export default NotFoundPage;
