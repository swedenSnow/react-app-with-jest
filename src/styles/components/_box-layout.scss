.box-layout {
    height:100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
    background: url(/images/bg.jpg);
    background-size: cover;
    font-family: $regular-font;
}

.box-layout__box{
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    background: fade-out($color: #fff, $amount: .35);
    border-radius: 5px;
    text-align: center;
    width: 30rem;
    padding: $l-size $m-size;
    
}

.box-layout__title{
    margin: 0 0 $m-size 0;
    line-height: 1;
    font-family: $italic-font;
    font-size: 4rem;
}
