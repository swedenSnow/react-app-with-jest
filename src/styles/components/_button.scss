.button{
    max-width: 230px;
    height: 50px;
    background: fade-out($blue, $amount: .15);
    position: relative;
    padding: 1rem;
    flex-grow: 1;
    margin: 0;
    border-radius: 8.5px;
    font-size: $font-size-large;
    font-weight: bold;
    color: #fff;
    border: none;
    margin-bottom: $s-size;

    &.button:hover{
        background:#34a853;
        box-shadow: 1px 1px 1px 1px $dark-grey;
        border: none;
        padding-right: 1rem;
        letter-spacing: .4px;
        padding-left: 0.5rem;
      }
    
}

.button:hover:after {
    opacity: 1;
    right: 0;
  }

.button:after {
    content: url(/images/sign-in-icon.png);
    position: absolute;
    opacity: 0;
    top: 0;
    right: 4rem;
    transition: 500ms;
    margin-right: 1.23rem;;
    }

.button__github {
    @extend .button;
    background: $dark-grey;

    &:hover {
        background: black;
    }

    &:after {
        content: none;
    }
}   

.button--link{
    display: inline-block;
    background: fade-out($blue, $amount: .15);
    border-radius: 8.5px;
    font-size: $font-size-large;
    font-weight: bold;
    color: #fff;
    border: none;
    padding: $s-size;
    text-decoration: none;
    line-height: 1;
    cursor: pointer;
    transition: all ease-in-out 200ms;

        &:hover{
            background: $dark-blue;
            box-shadow: 1px 1px 1px 1px $dark-grey;
            padding: $m-size;;
    }
}

.button--link__logout{
    color: #fff;
    background: none;
    border: none;
    font-size: $font-size-large;
    font-weight: 700;
    font-family: $regular-font;
    transition: all ease-in-out 200ms;
}

.button--link__logout:hover{
    color: $blue;
    background: none;
    border: none;
    // font-size: calc(100.5% + 0.5vw);
    font-size: 2rem;
}

.button--secondary {
    background: rgba(128, 127, 119, 0.247);
}