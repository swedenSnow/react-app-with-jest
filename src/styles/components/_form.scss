.form {
    display: flex;
    flex-flow: column wrap;
    > * {
        margin-bottom: $m-size;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-family: $italic-font;
    font-weight: 200;
    color: $dark-grey;
}

.form input,
textarea,
.SingleDatePickerInput {
    border-radius: 5px;
    outline-color: $deepAqua;
}
