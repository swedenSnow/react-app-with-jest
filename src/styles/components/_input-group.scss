.input-group{
    display: flex;
    flex-flow: column wrap;
    margin-bottom: $m-size;

    @media (min-width:$desktop-breakpoint) {
        flex-flow: row wrap;
        margin-bottom: $l-size;
        
    }
}

.input-group__item{
    margin-bottom: $s-size;

    @media (min-width:$desktop-breakpoint) {
        margin:0 $s-size 0 0;
        
    }
}