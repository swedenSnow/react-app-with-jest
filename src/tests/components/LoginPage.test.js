import React from 'react';
import { shallow, mount } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} />);
    wrapper.find('[data-test="gmail"]').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});

test('should call startLoginGit on button click', () => {
    const startLoginGit = jest.fn();
    const wrapper = shallow(<LoginPage startLoginGit={startLoginGit} />);
    wrapper.find('[data-test="github"]').simulate('click');
    expect(startLoginGit).toHaveBeenCalledTimes(1);
});
