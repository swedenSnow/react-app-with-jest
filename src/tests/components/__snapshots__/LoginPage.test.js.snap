// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Keep track of your Budget
    </h1>
    <p>
      It's time to get your expenses under control
    </p>
    <button
      className="button"
    >
      Sign in with
      <span
        className="btn__icon"
      />
    </button>
  </div>
</div>
`;
